diff --git a/src/libsass/src/json.cpp b/src/libsass/src/json.cpp
index f7d06e4..41ee2f1 100644
--- a/src/libsass/src/json.cpp
+++ b/src/libsass/src/json.cpp
@@ -1286,7 +1286,7 @@ static void emit_number(SB *out, double num)
    * like 0.3 -> 0.299999999999999988898 .
    */
   char buf[64];
-  sprintf(buf, "%.16g", num);
+  snprintf(buf, sizeof(buf), "%.16g", num);
 
   if (number_is_valid(buf))
     sb_puts(out, buf);
